ci/benchmark.sh

```diff
@@ -0,0 +1,45 @@
+#! /usr/bin/env bash
+
+ZEEK_BENCHMARK_ENDPOINT="/zeek"
+
+# Setting this causes any command failures to immediately cause the script to fail.
+set -e
+
+# Don't do this for any branch that isn't from the main zeek repo.
+# TODO: is it possible to do this from cirrus.yml instead of here?
+if [ "${CIRRUS_REPO_OWNER}" != "zeek" ]; then
+    echo "Benchmarks are skipped for repositories outside of the main Zeek project"
+    exit 0
+fi
+
+BUILD_URL="https://api.cirrus-ci.com/v1/artifact/build/${CIRRUS_BUILD_ID}/${CIRRUS_TASK_NAME}/upload_binary/build.tgz"
+
+# Generate an md5 hash of the build file. We can do this here because the path to the
+# file still exists from the prior scripts.
+BUILD_HASH=$(md5sum build.tgz | awk '{print $1}')
+
+# Generate an HMAC digest for the path plus a timestamp to send as an authentication
+# header. Openssl outputs a hex string here so there's no need to base64 encode it.
+# TODO: would it make sense to add the build hash as part of the hmac key here just
```

Comment:
Yeah, I think so, otherwise the MAC remains reusable for any content (unless there's something done with the timestamp?)

Reply:
The timestamp is used as part of the data to generate the HMAC digest, so it's at least unique that regard. Using the build hash would just make it a little more unique.

Reply:
I was thinking that if somebody somehow got access to one previous pair of `HMAC_DIGEST` and `TIMESTAMP`, they could reuse that to `POST` a different custom `build.tgz` to the server in the future. I might be missing something, but the only way I see how that could be prevented right now is if the server checked the timestamp as well and accepted it only if it where, like, within the most recent hour or so. If we included the MD5 into the HMAC, then it would be tied to the actual tgz under test and couldn't be reused for any other content.

(Would be good to switch to sha256 for the tgz hash then as well).